src/DependencyInjection/Compiler/PurgeTypeSettingsPass.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace EzSystems\PlatformHttpCacheBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+class PurgeTypeSettingsPass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container)
+    {
+        $purgeType = $container->getParameter('purge_type');
+        $purgeServiceId = null;
+        $responseConfiguratorServiceId = null;
+
+        $purgeClientServiceIds = $container->findTaggedServiceIds('ezplatform.http_cache.purge_client');
+        foreach ($purgeClientServiceIds as $purgeClientServiceId => $attributes) {
+            $currentPurgeTypeId = null;
+            $currentResponseConfiguratorId = null;
+            foreach ($attributes as $attribute) {
+                if (array_key_exists('purge_type', $attribute)) {
+                    $currentPurgeTypeId = $attribute['purge_type'];
+                }
+                if (array_key_exists('response_configurator', $attribute)) {
+                    $currentResponseConfiguratorId = $attribute['response_configurator'];
+                }
+                if ($currentPurgeTypeId !== null && $currentResponseConfiguratorId !== null) {
+                    if ($purgeType === $attribute['purge_type']) {
+                        $purgeServiceId = $purgeClientServiceId;
+                        $responseConfiguratorServiceId = $attribute['response_configurator'];
+                        break 2;
+                    }
+                }
+            }
+            if ($currentPurgeTypeId === null) {
+                throw new \InvalidArgumentException("Missing attribute 'purge_type' in tagged service '$purgeClientServiceId'.");
+            }
+            if ($currentResponseConfiguratorId === null) {
+                throw new \InvalidArgumentException("Missing attribute 'response_configurator' in tagged service '$purgeClientServiceId'.");
+            }
+        }
+
+        if ($purgeServiceId ===  null) {
+            throw new \InvalidArgumentException("No driver found being able to handle purge_type '$purgeType'.");
+        }
+
+        $container->setAlias('ezplatform.http_cache.purge_client', $purgeServiceId);
+
+        // If purge_type is not 'local' we must override ezpublish.http_cache.purge_client.fos too, which still uses
+        // eZ\Publish\Core\MVC\Symfony\Cache\Http\FOSPurgeClient
+        // Or else, that deprecated class is still used by cache:clear
+        if ($purgeType !== 'local') {
+            $container->setAlias('ezpublish.http_cache.purge_client', $purgeServiceId);
```

Comment:
should use *ezpublish.http_cache.purge_client* I think

Reply:
No, I meant what I wrote here....
This line ensured that correct purge client was used when running `php app/console clear:cache`

However, it seems like it was decided in https://github.com/ezsystems/ezplatform-http-cache/pull/15 that we should not clear http cache when doing a clear cache

So, if https://github.com/ezsystems/ezplatform-http-cache/pull/15 stays, this line can be removed

Reply:
15 was not necessarily the final say on it, it was to solve a bug where kernel purge system will be called on cache:clear, when if we need such logic should have the proper config for it here.

We don't clear Stash / Symfony cache either on cache:clear by default _(unless file system cache is used)_, but it should be possible somehow to do it when content or schema for some reason changes on deployment.

Reply:
But this line can indeed be removed, ca:cl is not supposed to call kernel purge client anymore with this bundle enabled. Actually almost all http cache code in kernel is deprecated in favor of this bundle.

Reply:
in any case are we sure we don't want to use `ezplatform.http_cache.purge_client` and not `ezpublish.http_cache.purge_client` ?

Reply:
I don't think I understood your last comment, but I have removed the line in question as it does not do anything after https://github.com/ezsystems/ezplatform-http-cache/pull/15 was merged